src/lua/LuaSpace.cpp

```diff
@@ -276,8 +276,9 @@ static int l_space_put_ship_on_route(lua_State *l)
 			const vector3d y = z.Cross(shippos).NormalizedSafe();
 			// just the third axis of this basis
 			const vector3d x = y.Cross(z);
+
 			// this is the basis in which the position of the ship will rotate
-			const matrix3x3d corrCS = matrix3x3d::FromVectors(x, y, z);
+			const matrix3x3d corrCS = matrix3x3d::FromVectors(x, y, z).Transpose();
```

Comment:
I'd appreciate a comment here denoting the fact that FromVectors is returning a matrix in the wrong (columnar) layout. Otherwise, this is good to merge.

Reply:
@Web-eWorks to be honest, I find it difficult to say whether this is correct or not, the question is, what does the action "construct a matrix3x3 from vector3s" mean?

Reply:
Matrix3x3::FromVectors (should, as an API contract) create a row-major orthonormal rotation matrix from three basis vectors defining the directions of the cartesian X, Y, Z axes under the transformation of the resulting matrix as long as all three vectors are unit-length and perpendicular to each other.

In this regard, returning a column-major matrix is invalid behavior, although given that transposition == inversion for affine transformation matrices, it's entirely possible that it has been miscommunicated and the function is returning the right matrix but the operation with the resulting matrix is required to be an inverse transformation.

Reply:
Looking at the code, it appears that the matrix is being constructed in the correct order; what is happening here is that FromVectors is creating a matrix that takes a coordinate system and rotates it into the 'world' coordinate system, but the code wants to perform calculations on world positions in the simplified coordinate system. As such, an inverse transformation is required to bring the world coordinates (targpos) into the coordinate system, and then the result in the simplified coordinate space is transformed back into world space (by an inverse-inverse transformation, which is the same as transforming by the result of FromVectors directly).

Reply:
This could be rewritten (without the initial .Transpose() call) using inverse-multiplication syntax, although I don't particularly recommend it:
```cpp
const vector3d safe1 = corrCS * (matrix3x3d::RotateY(+asin(erad / len)) * (-targpos * corrCS)).Normalized() * targdist;
```